### PR TITLE
set KUBECONFIG before argocd login

### LINKED ISSUE #125: `repo bootstrap` fails when running `argocd login` if passing different `--kubeconfig` argument

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: pkg/argocd/argocd.go [hunk 1]
```diff
@@ -103,13 +105,20 @@ func Login(opts *LoginOptions) error {
...
+		"--password",
+		opts.Password,
 		"--name",
 		"autopilot",
 	}
+	if opts.KubeConfig != "" {
+		origKubeConfig := os.Getenv("KUBECONFIG")
+		defer func() { os.Setenv("KUBECONFIG", origKubeConfig) }()
+		if err := os.Setenv("KUBECONFIG", opts.KubeConfig); err != nil {
+			return err
```
Comment (roi-codefresh) [inline pkg/argocd/argocd.go:119]: return a wrapped error here: `failed to set KUBECONFIG env var`, cause I'm not sure the error returned here would be clear

File: pkg/argocd/argocd.go [hunk 2]
```diff
@@ -31,9 +32,10 @@ type (
 	}
 
 	LoginOptions struct {
-		Namespace string
-		Username  string
-		Password  string
+		Namespace  string
+		Username   string
+		Password   string
+		KubeConfig string
```
Comment (roi-codefresh) [inline pkg/argocd/argocd.go:38]: what about kube context?

**Review by roi-codefresh — APPROVED** [review]

File: pkg/argocd/argocd.go (hunk 2 above)
Comment (noam-codefresh) [reply pkg/argocd/argocd.go:39]: removed most kube flags (except for timeout, kubeconfig and namespace).

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**